src/routes/auth.py

```diff
@@ -0,0 +1,60 @@
+import requests
+import os
+from flask import Blueprint, redirect, request
+from flask_jwt_extended import current_user, create_access_token
+
+from ..config import *
+from ..models import DiscordOAuth, User
+
+
+discord_bp = Blueprint("auth_bp",__name__, url_prefix='/discord/auth')
+
+@discord_bp.route("/login")
+def login():
+    """
+    Redirect to discord auth
+    """
+    return redirect(os.environ.get("DISCORD_OAUTH_URL"))
+
+@discord_bp.route("/login/callback")
+def callback():
+    """
+    Returned from discord after oauth
+    """
+
+    token_access_code = request.args.get("code",None)
+    data = {
+        "client_id": os.environ.get("DISCORD_CLIENT_ID"),
+        "client_secret": os.environ.get("DISCORD_CLIENT_SECRET"),
+        "grant_type": "authorization_code",
+        "code": token_access_code,
+        "redirect_uri": os.environ.get("DISCORD_OAUTH_REDIRECT"),
+    }
```

Comment:
A `state` parameter is recommended to prevent CSRF attacks. This can be any random string, but *may* be used to encode something meaningful to retrieve later.
The `state` parameter is returned as-is in the callback so you know that the callback was hit from an actual OAuth flow.

Reply:
70b5645 addresses these issues

Reply:
> [70b5645](https://github.com/IITM-BS-Codebase/iitm-backend/commit/70b5645f48113b1bf177bbbb5ca2090e2f5bd9ee) addresses these issues

It doesn't :pain:

I'll author a fix in a while 

The main issue is state shouldn't be global